crypto/crypto_aes_aesni.h

```diff
@@ -23,6 +29,17 @@ void * crypto_aes_key_expand_aesni(const uint8_t *, size_t);
 void crypto_aes_encrypt_block_aesni(const uint8_t[16], uint8_t[16],
     const void *);
 
+#ifdef CPUSUPPORT_X86_SSE2
```

Comment:
What do we want here?
- we only need SSE2 to use `__m128i`
- OTOH, there's no point defining this unless we have AESNI as well

I've gone back and forth a few times.  I currently have _SSE2, but I'm not very firm about that.  This is the first time we're adding cpu-specific functions to header files, so we don't have a precedent to look at.